### PR TITLE
Security Group Rule ICMP Type, Code -> Any

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMISSecurityGroupRule_basic'

```
Screenshot 2022-11-22 at 12 33 16 PM

```
resource "ibm_is_vpc" "testacc_vpc" {
  name = "sunitha"
}

resource "ibm_is_security_group" "testacc_security_group" {
  name = "sunitha-sg"
  vpc  = ibm_is_vpc.testacc_vpc.id
}

resource "ibm_is_security_group_rule" "testacc_security_group_rule_all" {
  group     = ibm_is_security_group.testacc_security_group.id
  direction = "inbound"
  remote    = "127.0.0.1"
}

resource "ibm_is_security_group_rule" "testacc_security_group_rule_icmp" {
  group     = ibm_is_security_group.testacc_security_group.id
  direction = "inbound"
  remote    = "127.0.0.1"
  icmp {
    code = 256
    type = 30
  }
}
resource "ibm_is_security_group_rule" "testacc_security_group_rule_icmp_any" {
  group     = ibm_is_security_group.testacc_security_group.id
  direction = "inbound"
  remote    = "127.0.0.1"
  icmp {
    # code = 256
    # type = 30
  }
}

resource "ibm_is_security_group_rule" "testacc_security_group_rule_tcp" {
  group     = ibm_is_security_group.testacc_security_group.id
  direction = "inbound"
  remote    = "127.0.0.1"
  tcp {
    # code = 256
    # type = 30
  }
}

data "ibm_is_security_group_rules" "example" {
  security_group = ibm_is_security_group.testacc_security_group.id
}

data "ibm_is_security_group_rule" "example" {
  security_group_rule = ibm_is_security_group_rule.testacc_security_group_rule_icmp.rule_id
  security_group      = ibm_is_security_group.testacc_security_group.id
}
```

### CODE REVIEW

**Comment by SunithaGudisagarIBM** [comment]
```
sunitha@Sunithas-MacBook-Pro vpc-ibm-lb-config % terraform show
# data.ibm_is_security_group_rule.example:
data "ibm_is_security_group_rule" "example" {
    code                = 256
    direction           = "inbound"
    href                = "https://us-south.iaas.cloud.ibm.com/v1/security_groups/r006-1f31afb0-416e-4ddc-9a04-1f87343aa8fd/rules/r006-0e257e96-ac9e-4861-921d-0823501cf025"
    id                  = "r006-0e257e96-ac9e-4861-921d-0823501cf025"
    ip_version          = "ipv4"
    protocol            = "icmp"
    remote              = [
        {
            address    = "127.0.0.1"
            cidr_block = ""
            crn        = ""
            deleted    = []
            href       = ""
            id         = ""
            name       = ""
        },
    ]
    security_group      = "r006-1f31afb0-416e-4ddc-9a04-1f87343aa8fd"
    security_group_rule = "r006-0e257e96-ac9e-4861-921d-0823501cf025"
    type                = 30
}
```

**Comment by SunithaGudisagarIBM** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4051

**Comment by SunithaGudisagarIBM** [comment]
```
sunitha@Sunithas-MacBook-Pro vpc-ibm-lb-config % terraform show
# data.ibm_is_security_group_rule.example:
data "ibm_is_security_group_rule" "example" {
    code                = 256
    direction           = "inbound"
    href                = "https://us-south.iaas.cloud.ibm.com/v1/security_groups/r006-1f31afb0-416e-4ddc-9a04-1f87343aa8fd/rules/r006-0e257e96-ac9e-4861-921d-0823501cf025"
    id                  = "r006-0e257e96-ac9e-4861-921d-0823501cf025"
    ip_version          = "ipv4"
    protocol            = "icmp"
    remote              = [
        {
            address    = "127.0.0.1"
            cidr_block = ""
            crn        = ""
            deleted    = []
            href       = ""
            id         = ""
            name       = ""
        },
    ]
    security_group      = "r006-1f31afb0-416e-4ddc-9a04-1f87343aa8fd"
    security_group_rule = "r006-0e257e96-ac9e-4861-921d-0823501cf025"
    type                = 30
}
```

File: ibm/service/vpc/resource_ibm_is_security_group_rule.go [hunk 1]
```diff
@@ -539,8 +545,8 @@ func parseIBMISSecurityGroupRuleDictionary(d *schema.ResourceData, tag string, s
 	sgTemplate := &vpcv1.SecurityGroupRulePrototype{}
 	sgTemplateUpdate := &vpcv1.UpdateSecurityGroupRuleOptions{}
 	var err error
-	parsed.icmpType = -1
-	parsed.icmpCode = -1
+	// parsed.icmpType = -1
```
Comment (ujjwal-ibm) [inline ibm/service/vpc/resource_ibm_is_security_group_rule.go:548]: remove commented code

File: ibm/service/vpc/resource_ibm_is_security_group_rule.go [hunk 2]
```diff
@@ -92,12 +92,14 @@ func ResourceIBMISSecurityGroupRule() *schema.Resource {
 						isSecurityGroupRuleType: {
 							Type:         schema.TypeInt,
 							Optional:     true,
+							Default:      255,
```
Comment (ujjwal-ibm) [inline ibm/service/vpc/resource_ibm_is_security_group_rule.go:95]: add test cases

**Review by ujjwal-ibm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**